### DISCUSSION TITLE
How to parameterise a project with options?

### DISCUSSION THREAD

**Discussion by anthonynorth** [ropensci/targets#657] [Help] [answered]
How can we supply options to a project, outside of the script?

I have a multi-project setup, where each project has _almost_ identical options set with `tar_option_set()`; I would like to supply these options as an input to `tar_make()`, similar to how we can supply project config.

**Comment by wlandau** [comment, marked as answer]
If you control the `TAR_PROJECT` environment variable through `.Renviron`, then you can write custom code to switch among options that way. The option-setting code could take the form of a script sourced from each of your project-level `_targets.R` files. I do not have plans for explicit built-in multi-project options.

**Reply by anthonynorth** [reply]
Thanks. 

This is similar to what I currently have, but I don't use the environment variable to set the project, I just read the yaml and apply over each project:

```r
yaml::read_yaml("_targets.yaml") |>
  lapply(do.call, what = targets::tar_make)
```

The project scripts themselves set project options by executing a partial wrapper over `tar_option_set()`.

**Reply by kkami1115** [reply]
Can I ask you to tell me more about that? I want to run the same pipeline with multiple files (paths), so I'm interested in your project management method.

**Reply by anthonynorth** [reply]
What we have is likely a pretty standard multi-project targets project. I think it's pretty close to what is described in https://books.ropensci.org/targets/projects.html#configure-each-project.

We have multiple projects that share a folder structure that is typical of project initialised with [`{tflow}`](https://github.com/milesmcbain/tflow), but with several project scripts (e.g. `project_a.R`, `project_b.R`) and config (`_targets.yaml`). We can run all projects with the snippet above if we need to.

The "sub-projects" are currently independent, so order of execution is irrelevant. We _may_ extract common config (i.e. targets that parameterise other targets) into a project which other projects can read from, but we haven't done this yet so it only seems like a good idea at this point -- perhaps better would be to just encapsulate this config in functions instead.

I hope that helps.
